includes/class-wc-embed.php

```diff
@@ -0,0 +1,160 @@
+ *
+ * @class 		WC_Embed
+ * @version		2.5.0
+ * @package		WooCommerce/Classes/Embed
+ * @category	Class
+ * @author 		WooThemes
+ */
+class WC_Embed {
+
+	/** @var WC_Embed The single instance of the class */
+	protected static $_instance = null;
+
+	/**
+	 * Main WC_Embed Instance.
+	 *
+	 * Ensures only one instance of WC_Embed is loaded or can be loaded.
+	 *
+	 * @since 2.5
+	 * @static
+	 * @return WC_Embed Main instance
+	 */
+	public static function instance() {
+		if ( is_null( self::$_instance ) ) {
+			self::$_instance = new self();
+		}
+		return self::$_instance;
+	}
+
+	/**
+	 * Cloning is forbidden.
+	 *
+	 * @since 2.5
+	 */
+	public function __clone() {
+		_doing_it_wrong( __FUNCTION__, __( 'Cheatin&#8217; huh?', 'woocommerce' ), '2.1' );
+	}
+
+	/**
+	 * Unserializing instances of this class is forbidden.
+	 *
+	 * @since 2.5
+	 */
+	public function __wakeup() {
+		_doing_it_wrong( __FUNCTION__, __( 'Cheatin&#8217; huh?', 'woocommerce' ), '2.1' );
+	}
+
+	/**
+	 * Init embed class.
+	 */
+	public static function init() {
+
+        // filter all of the content that's going to be embedded
+        add_filter( 'the_title', array( 'WC_Embed', 'the_title' ), 10 );
+        add_filter( 'the_excerpt_embed', array( 'WC_Embed', 'the_excerpt' ), 10 );
+
+        // make sure no comments display. Doesn't make sense for products
+        add_filter( 'get_comments_number', array( 'WC_Embed', 'get_comments_number' ), 10);
+        add_filter( 'comments_open', array( 'WC_Embed', 'comments_open' ), 10 );
+
+	}
+
+    /**
+     * Create the title for embedded products - we want to add the price to it
+     *
+     * @return string
+     */
+    public static function the_title( $title ) {
+        // make sure we're only affecting embedded products
+        if ( WC_Embed::is_embedded_product() ) {
+
+            // get product
+            $_pf = new WC_Product_Factory();
+            $_product = $_pf->get_product( get_the_ID() );
+
+            // add the price
+            $title = $title . '<span class="price" style="float: right;">' . $_product->get_price_html() . '</span>';
+        }
+        return $title;
+    }
+
+    /**
+     * Check if this is an embedded product - to make sure we don't mess up regular posts
+     *
+     * @return bool
+     */
+    public static function is_embedded_product() {
+        if ( function_exists( 'is_embed' ) && is_embed() && is_product() ) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Create the excerpt for embedded products - we want to add the buy button to it
+     *
+     * @return string
+     */
+    public static function the_excerpt( $excerpt ) {
+        //  make sure we're only affecting embedded products
+        if ( WC_Embed::is_embedded_product() ) {
+
+            // get product
+            $_pf = new WC_Product_Factory();
+            $_product = $_pf->get_product( get_the_ID() );
```

Comment:
Don't need t use `WC_Product_Factory`, you can just use `wc_get_product()`: https://github.com/woothemes/woocommerce/blob/753014da5a6df243928e4b7baebb035300ab70ff/includes/wc-product-functions.php#L24-L26